### ISSUE TITLE
Intermittent general exception on iRacing startup

### ISSUE THREAD

**Issue by simracer-cz** [simracer-cz/iFlag#2]
Users report an occasional general exception popup window:

```
See the end of this message for details on invoking 
just-in-time (JIT) debugging instead of this dialog box.

************** Exception Text **************
System.NullReferenceException: Object reference not set to an instance of an object.
   at iFlag.mainForm.matchSpotterOverlay() in C:\Users\pisi\Documents\iFlag-git\source\software\iFlag\overlays.cs:line 38
   at iFlag.mainForm.matchOverlays() in C:\Users\pisi\Documents\iFlag-git\source\software\iFlag\overlays.cs:line 27
   at iFlag.mainForm.showFlag(Int64 flagID) in C:\Users\pisi\Documents\iFlag-git\source\software\iFlag\dispatcher.cs:line 58
   at iFlag.mainForm.showDemoFlag() in C:\Users\pisi\Documents\iFlag-git\source\software\iFlag\dispatcher.cs:line 77
   at iFlag.mainForm.demoTimer_Tick(Object sender, EventArgs e) in C:\Users\pisi\Documents\iFlag-git\source\software\iFlag\mainForm.cs:line 163
   at System.Windows.Forms.Timer.OnTick(EventArgs e)
   at System.Windows.Forms.Timer.TimerNativeWindow.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)


************** Loaded Assemblies **************
mscorlib
    Assembly Version: 4.0.0.0
    Win32 Version: 4.7.3362.0 built by: NET472REL1LAST_C
    CodeBase: file:///C:/Windows/Microsoft.NET/Framework/v4.0.30319/mscorlib.dll
----------------------------------------
iflag
    Assembly Version: 0.63.0.0
    Win32 Version: 0.77.0.0
    CodeBase: file:///C:/Users/Josh/Documents/iFlag/iFlag.exe
----------------------------------------
System.Windows.Forms
    Assembly Version: 4.0.0.0
    Win32 Version: 4.7.3324.0 built by: NET472REL1LAST_C
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms/v4.0_4.0.0.0__b77a5c561934e089/System.Windows.Forms.dll
----------------------------------------
System
    Assembly Version: 4.0.0.0
    Win32 Version: 4.7.3362.0 built by: NET472REL1LAST_C
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System/v4.0_4.0.0.0__b77a5c561934e089/System.dll
----------------------------------------
System.Drawing
    Assembly Version: 4.0.0.0
    Win32 Version: 4.7.3190.0 built by: NET472REL1LAST_C
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Drawing/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Drawing.dll
----------------------------------------
iRSDKSharp
    Assembly Version: 1.0.0.0
    Win32 Version: 1.0.0.0
    CodeBase: file:///C:/Users/Josh/Documents/iFlag/iRSDKSharp.DLL
----------------------------------------
System.Configuration
    Assembly Version: 4.0.0.0
    Win32 Version: 4.7.3324.0 built by: NET472REL1LAST_C
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Configuration/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll
----------------------------------------
System.Core
    Assembly Version: 4.0.0.0
    Win32 Version: 4.7.3362.0 built by: NET472REL1LAST_C
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Core/v4.0_4.0.0.0__b77a5c561934e089/System.Core.dll
----------------------------------------
System.Xml
    Assembly Version: 4.0.0.0
    Win32 Version: 4.7.3190.0 built by: NET472REL1LAST_C
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
----------------------------------------

************** JIT Debugging **************
To enable just-in-time (JIT) debugging, the .config file for this
application or computer (machine.config) must have the
jitDebugging value set in the system.windows.forms section.
The application must also be compiled with debugging
enabled.

For example:

<configuration>
    <system.windows.forms jitDebugging="true" />
</configuration>

When JIT debugging is enabled, any unhandled exception
will be sent to the JIT debugger registered on the computer
rather than be handled by this dialog box.
```

The program is able to recover from this error when users use the _Continue_ button on the exception popup. Then it doesn't reappear.

**Comment by simracer-cz** [comment]
Investigations revealed that when iRacing SDK comes online and its `.IsConnected()` method returns `true`, for some moments after this the SDK actually doesn't have var headers in it yet )(`.VarHeaders.Count` returns zero) and attempts to work with the SDK in that time period (by querying its `.GetData()` for example) fail miserably with a null pointer reference exception which the users complained about.

Not sure if this behavior is be design from iRacing, or if it is a bug, but it does result in a racing condition which sometimes happen to go trough smoothly and sometimes it halts the execution.

The solution should be to update the logic places relying purely on `.IsConnected` to rely on already existing boolean `simConnected` flag, which becomes `true` only when:

```
sdk.IsConnected() && sdk.VarHeaders.Count > 0
```

**Comment by simracer-cz** [comment]
A unofficial patched version 0.80.1 is currently being tested by the bug reporters.

**Comment by simracer-cz** [comment]
Fixed in 4a9997b02b96afe2e3926c5bcd8fe2ad3d013796 and released in [v0.81](https://github.com/simracer-cz/iFlag/releases/tag/v0.81)
